app/lib/intervention.cjsx

```diff
@@ -0,0 +1,116 @@
+{getSessionID} = require '../lib/session'
+Dialog = require 'modal-form/dialog'
+{Link} = require 'react-router'
+
+Intervention = React.createClass
+
+  contextTypes:
+    geordi: React.PropTypes.object
+
+  propTypes:
+    experimentName: React.PropTypes.string.isRequired
+    sessionID: React.PropTypes.string.isRequired
+    interventionID: React.PropTypes.string.isRequired
+    interventionDetails: React.PropTypes.object.isRequired
+    disableInterventionFunction: React.PropTypes.func.isRequired
+
+  endIntervention: ->
+    experimentsClient.postDataToExperimentServer interventionMonitor,
+                                                 @context.geordi,
+                                                 @props.experimentName,
+                                                 @props.user.id,
+                                                 @props.sessionID,
+                                                 "intervention",
+                                                 @props.interventionID
+
+  skipIntervention: (event) ->
+    logData = {
+      sessID: @props.sessionID
+      ivnID: @props.interventionID
+      cancelled: true
+    }
+    experimentsClient.logExperimentData @context.geordi, 'skipIntervention', logData
+    @endIntervention()
+
+  answerQuestion: (event) ->
+    logData = {
+      sessID: @props.sessionID
+      ivnID: @props.interventionID
+      answer: document.getElementsByClassName("intervention-question")[0].value
+    }
+    experimentsClient.logExperimentData @context.geordi, 'interventionResponse', logData
+    @endIntervention()
+    event.target.disabled = true
+    event.preventDefault()
+
+  endStatement: (event) ->
+    @endIntervention()
+    event.target.disabled = true
+    event.preventDefault()
+
+  componentWillMount: ->
+    if experimentsClient.currentExperimentState isnt experimentsClient.EXPERIMENT_STATE_INTERVENTION_ON_SCREEN
+      experimentsClient.logExperimentState @context.geordi, interventionMonitor?.latestFromSugar, "interventionStart"
+
+  getBodyMarkup: (html) ->
+    { __html: html }
+
+  logLinkClick: (event) ->
+    logData = {
+      sessID: @props.sessionID
+      ivnID: @props.interventionID
+      linkAddress: event.target.getAttribute("href")
+    }
+    experimentsClient.logExperimentData @context.geordi, 'interventionLinkClicked', logData
+
+  optOut: ->
+    experimentsClient.optOutThisUser interventionMonitor, @context.geordi, @props.experimentName, @props.user.id
+    @props.disableInterventionFunction()
+
+  componentDidMount: ->
+    interventionBody = document.getElementsByClassName("intervention-body")[0]
+    linksInInterventions = interventionBody.getElementsByTagName("a")
+    for link in linksInInterventions
+      link.addEventListener("click", @logLinkClick)
+
+  render: ->
+    <Dialog style={maxWidth: '30%', paddingLeft: '1.2em', paddingRight: '1.2em', top: '20%'} tag="form" className="intervention" closeButton={true} onCancel={@skipIntervention}>
```

Comment:
Sorry, missed this. `maxWidth: 30%` is going to give a very thin dialog box on small screens. Maybe style this in the CSS and use media queries.